### ISSUE TITLE
Ok I give!

### ISSUE THREAD

**Issue by Warloxx01** [ACEmulator/ACE#3708]
### Bug Report or Feature Request (mark with an `x`)
```
- [x ] bug report -> please search issues before submitting
- [ ] feature request
```

### Repro steps.
Run Sever, Create a new Character, Talk to Jonathan, Get Academy Exit Token, Hand Token Back... Crash!

### The log given by the failure.
ACEmulator command prompt ready.

Type "acecommands" for help.

ACE >> 2021-10-17 05:58:12,492 INFO : [CHAT][AUDIT] [SYSTEM] says on the Audit channel, "World is now open"
2021-10-17 05:58:29,197 INFO : client owner connected with verified password
Unhandled exception. System.AggregateException: One or more errors occurred. (Object reference not set to an instance of an object.)
 ---> System.NullReferenceException: Object reference not set to an instance of an object.
   at ACE.Server.WorldObjects.WorldObject.ItemMagic(WorldObject target, Spell spell, WorldObject itemCaster, WorldObject weapon, Boolean equip) in D:\AC Stuff\0) Server\2020\ACE-1.31.4082\ACE-master\Source\ACE.Server\WorldObjects\WorldObject_Magic.cs:line 815
   at ACE.Server.WorldObjects.WorldObject.TryCastSpell_Inner(Spell spell, WorldObject target, WorldObject itemCaster, WorldObject weapon, Boolean isWeaponSpell, Boolean fromProc, Boolean tryResist, Boolean showMsg) in D:\AC Stuff\0) Server\2020\ACE-1.31.4082\ACE-master\Source\ACE.Server\WorldObjects\WorldObject_Magic.cs:line 121
   at ACE.Server.WorldObjects.WorldObject.TryCastSpell(Spell spell, WorldObject target, WorldObject itemCaster, WorldObject weapon, Boolean isWeaponSpell, Boolean fromProc, Boolean tryResist, Boolean showMsg) in D:\AC Stuff\0) Server\2020\ACE-1.31.4082\ACE-master\Source\ACE.Server\WorldObjects\WorldObject_Magic.cs:line 85
   at ACE.Server.WorldObjects.WorldObject.TryCastSpell_WithRedirects(Spell spell, WorldObject target, WorldObject itemCaster, WorldObject weapon, Boolean isWeaponSpell, Boolean fromProc, Boolean tryResist, Boolean showMsg) in D:\AC Stuff\0) Server\2020\ACE-1.31.4082\ACE-master\Source\ACE.Server\WorldObjects\WorldObject_Magic.cs:line 52
   at ACE.Server.WorldObjects.Managers.EmoteManager.ExecuteEmote(PropertiesEmote emoteSet, PropertiesEmoteAction emote, WorldObject targetObject) in D:\AC Stuff\0) Server\2020\ACE-1.31.4082\ACE-master\Source\ACE.Server\WorldObjects\Managers\EmoteManager.cs:line 243
   at ACE.Server.WorldObjects.Managers.EmoteManager.DoEnqueue(PropertiesEmote emoteSet, WorldObject targetObject, Int32 emoteIdx, PropertiesEmoteAction emote) in D:\AC Stuff\0) Server\2020\ACE-1.31.4082\ACE-master\Source\ACE.Server\WorldObjects\Managers\EmoteManager.cs:line 1628
   at ACE.Server.WorldObjects.Managers.EmoteManager.<>c__DisplayClass19_0.<Enqueue>b__1() in D:\AC Stuff\0) Server\2020\ACE-1.31.4082\ACE-master\Source\ACE.Server\WorldObjects\Managers\EmoteManager.cs:line 1611
   at ACE.Server.Entity.Actions.ActionEventDelegate.Act() in D:\AC Stuff\0) Server\2020\ACE-1.31.4082\ACE-master\Source\ACE.Server\Entity\Actions\ActionEventDelegate.cs:line 16
   at ACE.Server.Entity.Actions.ActionQueue.RunActions() in D:\AC Stuff\0) Server\2020\ACE-1.31.4082\ACE-master\Source\ACE.Server\Entity\Actions\ActionQueue.cs:line 33
   at ACE.Server.Entity.Landblock.TickMultiThreadedWork(Double currentUnixTime) in D:\AC Stuff\0) Server\2020\ACE-1.31.4082\ACE-master\Source\ACE.Server\Entity\Landblock.cs:line 441
   at ACE.Server.Managers.LandblockManager.<>c.<TickMultiThreadedWork>b__20_0(LandblockGroup landblockGroup) in D:\AC Stuff\0) Server\2020\ACE-1.31.4082\ACE-master\Source\ACE.Server\Managers\LandblockManager.cs:line 329
   at System.Threading.Tasks.Parallel.<>c__DisplayClass19_0`1.<ForWorker>b__1(RangeWorker& currentWorker, Int32 timeout, Boolean& replicationDelegateYieldedBeforeCompletion)
--- End of stack trace from previous location ---
   at System.Threading.Tasks.Parallel.<>c__DisplayClass19_0`1.<ForWorker>b__1(RangeWorker& currentWorker, Int32 timeout, Boolean& replicationDelegateYieldedBeforeCompletion)
   at System.Threading.Tasks.TaskReplicator.Replica.Execute()
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.TaskReplicator.Run[TState](ReplicatableUserAction`1 action, ParallelOptions options, Boolean stopOnFirstFailure)
   at System.Threading.Tasks.Parallel.ForWorker[TLocal](Int32 fromInclusive, Int32 toExclusive, ParallelOptions parallelOptions, Action`1 body, Action`2 bodyWithState, Func`4 bodyWithLocal, Func`1 localInit, Action`1 localFinally)
--- End of stack trace from previous location ---
   at System.Threading.Tasks.Parallel.ThrowSingleCancellationExceptionOrOtherException(ICollection exceptions, CancellationToken cancelToken, Exception otherException)
   at System.Threading.Tasks.Parallel.ForWorker[TLocal](Int32 fromInclusive, Int32 toExclusive, ParallelOptions parallelOptions, Action`1 body, Action`2 bodyWithState, Func`4 bodyWithLocal, Func`1 localInit, Action`1 localFinally)
   at System.Threading.Tasks.Parallel.ForEachWorker[TSource,TLocal](IEnumerable`1 source, ParallelOptions parallelOptions, Action`1 body, Action`2 bodyWithState, Action`3 bodyWithStateAndIndex, Func`4 bodyWithStateAndLocal, Func`5 bodyWithEverything, Func`1 localInit, Action`1 localFinally)
   at System.Threading.Tasks.Parallel.ForEach[TSource](IEnumerable`1 source, ParallelOptions parallelOptions, Action`1 body)
   at ACE.Server.Managers.LandblockManager.TickMultiThreadedWork() in D:\AC Stuff\0) Server\2020\ACE-1.31.4082\ACE-master\Source\ACE.Server\Managers\LandblockManager.cs:line 325
   at ACE.Server.Managers.LandblockManager.Tick(Double portalYearTicks) in D:\AC Stuff\0) Server\2020\ACE-1.31.4082\ACE-master\Source\ACE.Server\Managers\LandblockManager.cs:line 258
   at ACE.Server.Managers.WorldManager.UpdateGameWorld() in D:\AC Stuff\0) Server\2020\ACE-1.31.4082\ACE-master\Source\ACE.Server\Managers\WorldManager.cs:line 418
   at ACE.Server.Managers.WorldManager.UpdateWorld() in D:\AC Stuff\0) Server\2020\ACE-1.31.4082\ACE-master\Source\ACE.Server\Managers\WorldManager.cs:line 375
   at ACE.Server.Managers.WorldManager.<>c.<Initialize>b__15_0() in D:\AC Stuff\0) Server\2020\ACE-1.31.4082\ACE-master\Source\ACE.Server\Managers\WorldManager.cs:line 61
   at System.Threading.ThreadHelper.ThreadStart_Context(Object state)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location ---
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.ThreadHelper.ThreadStart()
2021-10-17 05:59:02,089 ERROR: System.AggregateException: One or more errors occurred. (Object reference not set to an instance of an object.)
 ---> System.NullReferenceException: Object reference not set to an instance of an object.
   at ACE.Server.WorldObjects.WorldObject.ItemMagic(WorldObject target, Spell spell, WorldObject itemCaster, WorldObject weapon, Boolean equip) in D:\AC Stuff\0) Server\2020\ACE-1.31.4082\ACE-master\Source\ACE.Server\WorldObjects\WorldObject_Magic.cs:line 815
   at ACE.Server.WorldObjects.WorldObject.TryCastSpell_Inner(Spell spell, WorldObject target, WorldObject itemCaster, WorldObject weapon, Boolean isWeaponSpell, Boolean fromProc, Boolean tryResist, Boolean showMsg) in D:\AC Stuff\0) Server\2020\ACE-1.31.4082\ACE-master\Source\ACE.Server\WorldObjects\WorldObject_Magic.cs:line 121
   at ACE.Server.WorldObjects.WorldObject.TryCastSpell(Spell spell, WorldObject target, WorldObject itemCaster, WorldObject weapon, Boolean isWeaponSpell, Boolean fromProc, Boolean tryResist, Boolean showMsg) in D:\AC Stuff\0) Server\2020\ACE-1.31.4082\ACE-master\Source\ACE.Server\WorldObjects\WorldObject_Magic.cs:line 85
   at ACE.Server.WorldObjects.WorldObject.TryCastSpell_WithRedirects(Spell spell, WorldObject target, WorldObject itemCaster, WorldObject weapon, Boolean isWeaponSpell, Boolean fromProc, Boolean tryResist, Boolean showMsg) in D:\AC Stuff\0) Server\2020\ACE-1.31.4082\ACE-master\Source\ACE.Server\WorldObjects\WorldObject_Magic.cs:line 52
   at ACE.Server.WorldObjects.Managers.EmoteManager.ExecuteEmote(PropertiesEmote emoteSet, PropertiesEmoteAction emote, WorldObject targetObject) in D:\AC Stuff\0) Server\2020\ACE-1.31.4082\ACE-master\Source\ACE.Server\WorldObjects\Managers\EmoteManager.cs:line 243
   at ACE.Server.WorldObjects.Managers.EmoteManager.DoEnqueue(PropertiesEmote emoteSet, WorldObject targetObject, Int32 emoteIdx, PropertiesEmoteAction emote) in D:\AC Stuff\0) Server\2020\ACE-1.31.4082\ACE-master\Source\ACE.Server\WorldObjects\Managers\EmoteManager.cs:line 1628
   at ACE.Server.WorldObjects.Managers.EmoteManager.<>c__DisplayClass19_0.<Enqueue>b__1() in D:\AC Stuff\0) Server\2020\ACE-1.31.4082\ACE-master\Source\ACE.Server\WorldObjects\Managers\EmoteManager.cs:line 1611
   at ACE.Server.Entity.Actions.ActionEventDelegate.Act() in D:\AC Stuff\0) Server\2020\ACE-1.31.4082\ACE-master\Source\ACE.Server\Entity\Actions\ActionEventDelegate.cs:line 16
   at ACE.Server.Entity.Actions.ActionQueue.RunActions() in D:\AC Stuff\0) Server\2020\ACE-1.31.4082\ACE-master\Source\ACE.Server\Entity\Actions\ActionQueue.cs:line 33
   at ACE.Server.Entity.Landblock.TickMultiThreadedWork(Double currentUnixTime) in D:\AC Stuff\0) Server\2020\ACE-1.31.4082\ACE-master\Source\ACE.Server\Entity\Landblock.cs:line 441
   at ACE.Server.Managers.LandblockManager.<>c.<TickMultiThreadedWork>b__20_0(LandblockGroup landblockGroup) in D:\AC Stuff\0) Server\2020\ACE-1.31.4082\ACE-master\Source\ACE.Server\Managers\LandblockManager.cs:line 329
   at System.Threading.Tasks.Parallel.<>c__DisplayClass19_0`1.<ForWorker>b__1(RangeWorker& currentWorker, Int32 timeout, Boolean& replicationDelegateYieldedBeforeCompletion)
--- End of stack trace from previous location ---
   at System.Threading.Tasks.Parallel.<>c__DisplayClass19_0`1.<ForWorker>b__1(RangeWorker& currentWorker, Int32 timeout, Boolean& replicationDelegateYieldedBeforeCompletion)
   at System.Threading.Tasks.TaskReplicator.Replica.Execute()
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.TaskReplicator.Run[TState](ReplicatableUserAction`1 action, ParallelOptions options, Boolean stopOnFirstFailure)
   at System.Threading.Tasks.Parallel.ForWorker[TLocal](Int32 fromInclusive, Int32 toExclusive, ParallelOptions parallelOptions, Action`1 body, Action`2 bodyWithState, Func`4 bodyWithLocal, Func`1 localInit, Action`1 localFinally)
--- End of stack trace from previous location ---
   at System.Threading.Tasks.Parallel.ThrowSingleCancellationExceptionOrOtherException(ICollection exceptions, CancellationToken cancelToken, Exception otherException)
   at System.Threading.Tasks.Parallel.ForWorker[TLocal](Int32 fromInclusive, Int32 toExclusive, ParallelOptions parallelOptions, Action`1 body, Action`2 bodyWithState, Func`4 bodyWithLocal, Func`1 localInit, Action`1 localFinally)
   at System.Threading.Tasks.Parallel.ForEachWorker[TSource,TLocal](IEnumerable`1 source, ParallelOptions parallelOptions, Action`1 body, Action`2 bodyWithState, Action`3 bodyWithStateAndIndex, Func`4 bodyWithStateAndLocal, Func`5 bodyWithEverything, Func`1 localInit, Action`1 localFinally)
   at System.Threading.Tasks.Parallel.ForEach[TSource](IEnumerable`1 source, ParallelOptions parallelOptions, Action`1 body)
   at ACE.Server.Managers.LandblockManager.TickMultiThreadedWork() in D:\AC Stuff\0) Server\2020\ACE-1.31.4082\ACE-master\Source\ACE.Server\Managers\LandblockManager.cs:line 325
   at ACE.Server.Managers.LandblockManager.Tick(Double portalYearTicks) in D:\AC Stuff\0) Server\2020\ACE-1.31.4082\ACE-master\Source\ACE.Server\Managers\LandblockManager.cs:line 258
   at ACE.Server.Managers.WorldManager.UpdateGameWorld() in D:\AC Stuff\0) Server\2020\ACE-1.31.4082\ACE-master\Source\ACE.Server\Managers\WorldManager.cs:line 418
   at ACE.Server.Managers.WorldManager.UpdateWorld() in D:\AC Stuff\0) Server\2020\ACE-1.31.4082\ACE-master\Source\ACE.Server\Managers\WorldManager.cs:line 375
   at ACE.Server.Managers.WorldManager.<>c.<Initialize>b__15_0() in D:\AC Stuff\0) Server\2020\ACE-1.31.4082\ACE-master\Source\ACE.Server\Managers\WorldManager.cs:line 61
   at System.Threading.ThreadHelper.ThreadStart_Context(Object state)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location ---
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.ThreadHelper.ThreadStart()

### Mention any other details that might be useful.
<!-- Please include any information that can help clarify the request here --
[Log.txt](https://github.com/ACEmulator/ACE/files/7359659/Log.txt)
>

I have Started a completely new Server, from scratch. Server Build (ACE-1.31.4082 ) World Build (ACE-World-16PY-db-v0.8.0) Patch Build ( ACE-World-Database-v0.9.231) . Only change made to anything was the config.js (server name, DB auto download thats about it for that file ) and to the ACE-World-Database-v0.9.231 file (line 1090 from v0.7.18 to v0.8.0 ) to reflect the true DB installed. Also was doing this to me in Sever Build ACE-1.31.4081 but I didnt have time to post... Moving :(

**Comment by Warloxx01** [comment]
Nope No Mods as stated above. Clean install. Not sure how to run in debug mode.

**Comment by ziang4891** [comment]
It does not do that for me, and I am running the latest build and Patches.  Since this isn't happening to others, there is something about what you are running.  Requires loading it into VS and running it there.

**Comment by LtRipley36706** [comment]
Re-produced, will be fixed with #3709

**Comment by Warloxx01** [comment]
Cool, Thanks LtRipley. Was starting to think I had a screw lose... Well maybe ;) Again thanks for looking into this. Thank you guys so much for all the HARD work and time you have vested in this project. All of you ROCK!!!
